tools/sz_repo_cli/lib/src/commands/src/deploy_ios_command.dart

```diff
@@ -0,0 +1,254 @@
+      )
+      ..addOption(
+        issuerIdOptionName,
+        help:
+            'The App Store Connect API Key Issuer ID used to authenticate. This can be found in the App Store Connect Developer Portal. Learn more at https://developer.apple.com/documentation/appstoreconnectapi/creating_api_keys_for_app_store_connect_api. If the parameter is not provided, the value of the APP_STORE_CONNECT_ISSUER_ID environment variable will be used. If no value is set, the deployment will fail. Example value: 00000000-0000-0000-0000-000000000000',
+      )
+      ..addOption(
+        privateKeyOptionName,
+        help:
+            'The App Store Connect API private key used for JWT authentication to communicate with Apple services. This can be found in the App Store Connect Developer Portal. Learn more at https://developer.apple.com/documentation/appstoreconnectapi/creating_api_keys_for_app_store_connect_api. If not provided, the key will be searched from the following directories in sequence for a private key file with the name AuthKey_<key_identifier>.p8: private_keys, ~/private_keys, ~/.private_keys, ~/.appstoreconnect/private_keys, where <key_identifier> is the value of --key-id. If not given, the value will be checked from the environment variable APP_STORE_CONNECT_PRIVATE_KEY. If no value is set, the deployment will fail.',
+      )
+      ..addOption(
+        whatsNewOptionName,
+        help:
+            "Release notes either for TestFlight or App Store review submission. Describe what's new in this version of your app, such as new features, improvements, and bug fixes. The string should not exceed 4000 characters. Example usage: --whats-new 'Bug fixes and performance improvements.'",
+      )
+      ..addOption(
+        flavorOptionName,
+        allowed: _iosFlavors,
+        help: 'The flavor to build for. Only the "prod" flavor is supported.',
+        defaultsTo: 'prod',
+      );
+  }
+
+  static const privateKeyOptionName = 'private-key';
+  static const keyIdOptionName = 'key-id';
+  static const issuerIdOptionName = 'issuer-id';
+  static const releaseStageOptionName = 'stage';
+  static const flavorOptionName = 'flavor';
+  static const whatsNewOptionName = 'whats-new';
+
+  @override
+  String get description =>
+      'Deploys the Sharezone iOS app to the App Store or TestFlight. Automatically bumps the build number and builds the app. Codemagic CLI tools are required.';
+
+  @override
+  String get name => 'ios';
+
+  @override
+  Future<void> run() async {
+    _throwIfFlavorIsNotSupportForDeployment();
+
+    // Is used so that runProcess commands print the command that was run. Right
+    // now this can't be done via an argument.
+    //
+    // This workaround should be addressed in the future.
+    isVerbose = true;
+
+    final buildNumber = await _getNextBuildNumber();
+    await _buildApp(buildNumber: buildNumber);
+    await _publish();
+
```

Comment:
Nice, very apprehensible code!

Nit: order the method implementations below in the same order (so that after `_throwIfFlavorIsNotSupportForDeployment` below comes `_getNextBuildNumber`, then `_buildApp`, then `_publish`). This helps to understand the code more quickly becaues you're going with the flow of the code when scrolling down in the IDE (looking at each method implementation in the order it will be executed)